git-cliff/src/lib.rs

```diff
@@ -559,40 +557,23 @@ pub fn run(mut args: Opt) -> Result<()> {
 			return Ok(());
 		};
 		if args.bumped_version {
-			if let Some(path) = args.output {
-				let mut output = File::create(path)?;
-				output.write_all(next_version.as_bytes())?;
-			} else {
-				println!("{next_version}");
-			}
+			writeln!(out, "{next_version}")?;
 			return Ok(());
 		}
 	}
+
 	if args.context {
-		return if let Some(path) = args.output {
-			let mut output = File::create(path)?;
-			changelog.write_context(&mut output)
-		} else {
-			changelog.write_context(&mut io::stdout())
-		};
+		changelog.write_context(&mut out)?;
+		return Ok(());
 	}
-	if let Some(ref path) = args.prepend {
-		changelog.prepend(fs::read_to_string(path)?, &mut File::create(path)?)?;
+
+	if let Some(path) = &args.prepend {
+		let mut out = io::BufWriter::new(File::create(path)?);
+		changelog.prepend(fs::read_to_string(path)?, &mut out)?;
```

Comment:
Okay something is still wrong here. `cargo run -- -u -p CHANGELOG.md` overwrites the entire changelog content with the unreleased section instead of prepending to the top.

Maybe revert the changes after `// Print the result.` and compare the result to see if it is caused by this change or not. Like I said, this part is extremely fragile (not tested very well) so it might be better to not touch it 😅  I'm happy to merge this PR with the other changes. (and I don't want to block #711 any further)

Reply:
Sorry about that, fixed in b6d2a219c64cad45f3d659826c961fa16b957b0e. I am pretty confident the code is equivalent now, comparing side by side with <https://github.com/orhun/git-cliff/blob/e7807e13c4b38aaa4a735ff05b69fdd6b57a7a85/git-cliff/src/lib.rs#L573-L599>
- the `output` variable is extracted
- the second `if args.context` is removed as it's unreachable
- after these changes the last if block becomes `if args.output.is_some() { generate } else if args.prepend.is_none() { generate }`, which is collapsed into one

The only behavior change I can see here is there is an extra new line in `bumped_version` with `output` specified (now using `writeln!` for both cases instead of `write_all`), I can revert this if you'd like https://github.com/orhun/git-cliff/blob/e7807e13c4b38aaa4a735ff05b69fdd6b57a7a85/git-cliff/src/lib.rs#L563-L571

Reply:
Looks good & works as expected now, thanks!